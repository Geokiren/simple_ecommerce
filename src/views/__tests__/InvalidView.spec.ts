import { mount } from '@vue/test-utils';
import { describe, expect, it } from 'vitest';

import InvalidView from '../InvalidView.vue';

describe('NotFoundPage', () => {
  it('renders the SVG icon', () => {
    const wrapper = mount(InvalidView);

    const svg = wrapper.find('svg');

    expect(svg.exists()).toBe(true);
  });

  it('renders the heading', () => {
    const wrapper = mount(InvalidView);

    const heading = wrapper.find('h1');

    expect(heading.exists()).toBe(true);
    expect(heading.text()).toBe('Oops! Page not found...');
  });

  it('renders the Go Home button', () => {
    const wrapper = mount(InvalidView);

    const button = wrapper.find('button');

    expect(button.exists()).toBe(true);
    expect(button.text()).toBe('Go Home');
  });

  it('matches snapshot', () => {
    const wrapper = mount(InvalidView);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
