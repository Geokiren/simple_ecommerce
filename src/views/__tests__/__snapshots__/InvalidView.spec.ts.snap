// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFoundPage > matches snapshot 1`] = `
"<section class="flex flex-col gap-4 items-center justify-center h-screen fixed top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2"><svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" class="size-52">
    <path stroke-linecap="round" stroke-linejoin="round" d="M12 9v3.75m9-.75a9 9 0 1 1-18 0 9 9 0 0 1 18 0Zm-9 3.75h.008v.008H12v-.008Z"></path>
  </svg>
  <h1 class="text-3xl font-bold">Oops! Page not found...</h1><button class="p-4 bg-active text-lg rounded text-alttext">Go Home</button>
</section>"
`;
